### PR TITLE
Add autocomplete attributes on FO login form

### LINKED ISSUE #14506: Browser console suggestion for the password form
**Describe the bug**
What do you think about this suggestion in console? Should we need to implement it in the code?

**To Reproduce**
1. Go to 'Sign in/Sign Up' page
2. Open browser consolve
3. See suggestion

**Screenshots**
Untitled

**Additionnal information**
PrestaShop version: PrestaShop 1.7.6 RC

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Autocomplete is enabled by default on many browsers, but it can provide warnings or fill wrong information, this is better to force them.
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #14506.
| How to test?  | Go on login form, log for a first time if you want, then see if chrome provide you an autocomplete (if you saved you'r mail/password inside it (you may already refused that years ago, this means that you won't have an autocomplete), also a lot of browsers have this by default, but it display a warning if you doesn't have it, it may change nothing exept kicking the warning

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21425)

### CODE REVIEW

File: classes/form/FormField.php [hunk 1]
```diff
@@ -181,4 +183,16 @@ public function getConstraints()
     {
         return $this->constraints;
     }
+
+    public function setAutocomplete($autocomplete)
```
Comment (matks) [inline classes/form/FormField.php]:
```suggestion
    public function setAutocompleteAttribute($autocomplete)
```

Suggestion: be as accurate as possible in function name to make sure people understand what it is. With 'autocomplete' alone someone could think it's, for example, a feature that performs ajax queries to auto-complete what you type 😄

Comment (matks) [reply classes/form/FormField.php]: Also @Progi1984 will tell you to type-hint this if I am not wrong 😛

File: classes/form/FormField.php [hunk 2]
```diff
@@ -181,4 +183,16 @@ public function getConstraints()
...
+    {
+        $this->autocomplete = $autocomplete;
+
+        return $this;
+    }
+
+    public function getAutocomplete()
```
Comment (matks) [inline classes/form/FormField.php]: @Progi1984 will tell you to type-hint the return type, I'm sure 😄 
```suggestion
    public function getAutocompleteAttribute(): string
```

File: themes/classic/templates/_partials/form-fields.tpl [hunk 3]
```diff
@@ -138,6 +138,7 @@
               name="{$field.name}"
               title="{l s='At least 5 characters long' d='Shop.Forms.Help'}"
               type="password"
+              {if $field.autocomplete}autocomplete="{$field.autocomplete}"{/if}
```
Comment (matks) [inline themes/classic/templates/_partials/form-fields.tpl:141]: Should'nt we use Smarty `escape` function ?

And I know this is not the scope of this PR but maybe we should (in another PR) apply the `escape` to EVERY other value in this TPL 😅 

Wdyt @PierreRambaud ?

**Review by matks — COMMENTED** [review]
Hi @NeOMakinG nice PHP PR 😄 

PR is mainly good, I provided some suggestions to improve the details but there are no major issues in it

File: classes/form/FormField.php (hunk 2 above)
Comment (NeOMakinG) [reply classes/form/FormField.php]: What about every other fields which are not typed at all ?

I'm not in favor of typing only on method while every siblings methods are not typed at all, that should be a complete refacto don't you think?

File: themes/classic/templates/_partials/form-fields.tpl (hunk 3 above)
Comment (NeOMakinG) [reply themes/classic/templates/_partials/form-fields.tpl:141]: Waiting for @PierreRambaud advice on this

File: classes/form/FormField.php (hunk 2 above)
Comment (matthieu-rolland) [reply classes/form/FormField.php]: I understand the will for homogeneity, but on a practical aspect, what would be the true downside of type-hinting the return type of new methods we add in legacy classes ?

**Comment by NeOMakinG** [comment]
@matks Did type methods I added, but should I type methods already defined before this PR ? I think this could be a BC

File: classes/form/FormField.php [hunk 4]
```diff
@@ -34,6 +34,7 @@ class FormFieldCore
     private $maxLength = null;
     private $errors = [];
     private $constraints = [];
+    private $autocomplete = '';
```
Comment (Progi1984) [inline classes/form/FormField.php:40]:
```suggestion
    /**
     * @var string
     */
    private $autocomplete = '';
```

File: classes/form/FormField.php [hunk 5]
```diff
@@ -181,4 +183,16 @@ public function getConstraints()
     {
         return $this->constraints;
     }
+
+    public function setAutocompleteAttribute($autocomplete): FormFieldCore
```
Comment (Progi1984) [inline classes/form/FormField.php]:
```suggestion
    /**
     * @param string $autocomplete
     *
     * @return FormFieldCore
     */
    public function setAutocompleteAttribute(string $autocomplete): FormFieldCore
```

**Review by Progi1984 — COMMENTED** [review]
Small improvemnts

File: classes/form/FormField.php [hunk 6]
```diff
@@ -181,4 +186,16 @@ public function getConstraints()
...
     }
+
+    public function setAutocompleteAttribute(string $autocomplete): FormFieldCore
+    {
+        $this->autocomplete = $autocomplete;
+
+        return $this;
+    }
+
+    public function getAutocompleteAttribute(): string
```
Comment (Progi1984) [inline classes/form/FormField.php]:
```suggestion
    /**
     * @param string $autocomplete
     *
     * @return FormFieldCore
     */
    public function setAutocompleteAttribute(string $autocomplete): FormFieldCore
    {
        $this->autocomplete = $autocomplete;

        return $this;
    }

    /**
     * @return string
     */
    public function getAutocompleteAttribute(): string
```

**Review by Progi1984 — APPROVED** [review]

File: themes/classic/templates/_partials/form-fields.tpl (hunk 3 above)
Comment (PierreRambaud) [reply themes/classic/templates/_partials/form-fields.tpl:141]: AFAIK, smarty already automatically escape html
`$smarty->escape_html = true;`
https://www.smarty.net/docs/en/variable.escape.html.tpl

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1gL6FydS2jVxbVKkNjxuSEVdjaYbwUkqm/view?usp=sharing
Without PR:
image

Thanks!

**Pull request merged**